classes/UserReputation.php

```diff
@@ -287,24 +341,34 @@ public function calcArenaReputation($player_level, $opponent_level) {
         }
     }
     // Calculate and return reputation gains/losses from pvp wins/losses
-    public function calcPvpRep($player_level, $player_rep_rank, $opponent_level, $opponent_rep_rank, $opponent_user_id, $winner = true) {
+    public function calcPvpRep($player_level, $player_rep_rank, $opponent_level, $opponent_rep_rank, $opponent_user_id, $winner = true): int {
         $player_levels_above_opponent = $player_level - $opponent_level;
         $rep_rank_difference = $player_rep_rank - $opponent_rep_rank;
         $rep_gain = 0;
 
-        // Set current kill
-        $this->recent_players_killed_ids_array[$opponent_user_id][] = time();
-        // Get kill count
-        $kill_count = isset($this->recent_players_killed_ids_array[$opponent_user_id]) ? sizeof($this->recent_players_killed_ids_array[$opponent_user_id]) : 0;
-        // Encode and set last pvp kills
-        $this->encodePvpKills();
+        // Pvp reputation disabled
+        if(!self::PVP_REP_ENABLED) {
+            return 0;
+        }
 
-        if($winner == true) {
+        if($winner) {
+            // Set current kill
+            $this->recent_players_killed_ids_array[$opponent_user_id][] = time();
```

Comment:
If this is doing mutations we should consider renaming, maybe instead of `calcPvPRep` it could be `handlePvPWin` and we handle adding/deducting in this method